src/Our.Umbraco.Ditto/ComponentModel/TypeConverters/HtmlStringConverter.cs

```diff
@@ -51,27 +52,34 @@ public override object ConvertFrom(ITypeDescriptorContext context, CultureInfo c
                 return null;
             }
 
-            var s = value.ToString();
-            if (value is HtmlString)
+            if (value is string)
             {
-                if (!string.IsNullOrWhiteSpace(s))
+                var text = value.ToString();
+
+                if (!string.IsNullOrWhiteSpace(text))
                 {
-                    s = TemplateUtilities.ParseInternalLinks(s);
+                    var umbracoHelper = ConverterHelper.UmbracoHelper;
+                    text = umbracoHelper.ReplaceLineBreaksForHtml(text);
                 }
 
-                return new HtmlString(s);
+                return new HtmlString(text);
             }
 
-            // This lets us convert textbox multiple data type properties to HtmlString's 
-            if (value is string)
+            if (value is HtmlString)
             {
-                if (!string.IsNullOrWhiteSpace(s))
+                var html = value.ToString();
+
+                if (!string.IsNullOrWhiteSpace(html))
                 {
-                    var umbracoHelper = ConverterHelper.UmbracoHelper;
-                    s = umbracoHelper.ReplaceLineBreaksForHtml(s);
+                    html = TemplateUtilities.ParseInternalLinks(html);
                 }
 
-                return new HtmlString(s);
+                return new HtmlString(html);
+            }
+
+            if (value is DynamicXml)
+            {
+                return ((DynamicXml)value).ToHtml();
```

Comment:
Not sure, I'll review my original code - see what happens with the property data.